pkgs/development/tools/misc/tideways/php-probe.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenvNoCC
+, lib
+, fetchurl
+, autoPatchelfHook
+, php
+}:
+
+let
+  soFile = {
+    "8.0" = "tideways-php-8.0.so";
+    "8.1" = "tideways-php-8.1.so";
+    "8.2" = "tideways-php-8.2.so";
+  }.${lib.versions.majorMinor php.version} or (throw "Unsupported PHP version.");
+in
+stdenvNoCC.mkDerivation rec {
+  pname = "tideways-php";
+  extensionName = "tideways";
+  version = "5.5.18";
+
+  src = passthru.sources.${stdenvNoCC.hostPlatform.system} or (throw "Unsupported platform for tideways-cli: ${stdenvNoCC.hostPlatform.system}");
+
+  nativeBuildInputs = lib.optionals stdenvNoCC.isLinux [
+    autoPatchelfHook
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D ${soFile} $out/lib/php/extensions/tideways.so
+
+    runHook postInstall
+  '';
+
+  passthru = {
```

Comment:
Why defining the sources in `passthru` instead of defining them in a `let/in` clause outside of the scope of the derivation ?

Reply:
`passthru` makes it convenient to build when updating in an update script, see e.g. https://github.com/NixOS/nixpkgs/blob/9f9fcc1483fb041f4d8708a78d025614d7a03430/pkgs/development/web/bun/default.nix

Reply:
Thanks!